### PR TITLE
Add License Information to EML

### PR DESCRIPTION
**Pull request by ThomasThelen**
We need to let the user pick a license for their package. This adds a parameter to the package endpoint. The licenses are keyed in constants.py

Most of the change can be read by first looking at the changes in constants.py, then going to repository.py and following the call stack. You'll see that the license id persists up to EML generation.

Note that in dataone_package.py I re-arranged some of the EML generation function. Mostly putting the code below the function declarations.
This corresponds to issue #128

### CODE REVIEW

File: server/constants.py [hunk 1]
```diff
@@ -79,3 +78,74 @@ class ExtraFileNames:
...
+    def text_from_id(id):
+        """
+        Given an ID, return the license text
+        :param id: The ID of the license
+        :type id: str
+        :return: The text of the corresponding license
+        :rtype: str
+        """
+        id = int(id)
+        if id == 0:
```
Comment (Xarthisius) [inline server/constants.py]: Why not use `Licence.LicenseIds.CC0` instead? It would be more readable IMHO.

File: server/constants.py [hunk 2]
```diff
@@ -79,3 +78,74 @@ class ExtraFileNames:
     """
     # Name for the tale config file
     tale_config = 'tale.yml'
+
+
+class Licence:
```
Comment (Xarthisius) [inline server/constants.py]: Can we use the same spelling (License)?

File: server/constants.py [hunk 3]
```diff
@@ -79,3 +78,74 @@ class ExtraFileNames:
...
+
+    class LicenseIds:
+        """
+        Key the licenses off the integers, starting at 0
+        """
+        CC0 = 0
+        CCBY3 = 1
+        CCBY4 = 2
+
+    class LicenseText:
```
Comment (Xarthisius) [inline server/constants.py]: Can we just convert it to dictionary? `text_from_id` would be unnecessary then.

File: server/dataone_package.py [hunk 4]
```diff
@@ -192,12 +216,13 @@ def add_object_record(name, description, size, object_format):
         add_object_record(item['name'], item['description'], item['size'], object_format)
 
     # Add a section for the tale.yml file
-    if file_sizes.get('tale_yaml') is not int() or None:
+    if file_sizes.get('tale_yaml') != int() or None:
```
Comment (Xarthisius) [inline server/dataone_package.py]: I'd use `not isinstance(file_sizes.get('tale_yaml'), int)`

File: server/dataone_package.py [hunk 5]
```diff
@@ -192,12 +216,13 @@ def add_object_record(name, description, size, object_format):
...
         description = "Configuration file that has information that will be useful " \
                       "for re-creating the computational environment."
         name = ExtraFileNames.tale_config
         object_format = 'application/x-yaml'
-        add_object_record(name, description, file_sizes.get('external_files'), object_format)
+        logger.info(file_sizes)
```
Comment (Xarthisius) [inline server/dataone_package.py]: Could you either remove it or make it more descriptive?

File: server/dataone_upload.py [hunk 6]
```diff
@@ -391,7 +403,7 @@ def create_upload_package(item_ids, tale, user, repository, jwt):
                                                                   item_ids,
                                                                   user,
                                                                   client)
-
+        logger.info(license_id)
```
Comment (Xarthisius) [inline server/dataone_upload.py]: Could you either remove it or make it more descriptive?

File: server/dataone_package.py (hunk 4 above)
Comment (ThomasThelen) [reply server/dataone_package.py]: Ah since get can return None, it can check both at once. Nice.

**Pull request merged**